packages/core/test/rehydration.test.ts

```diff
@@ -98,4 +98,21 @@ describe('rehydration', () => {
       expect(actual).toEqual(['active', 'root']);
     });
   });
+
+  describe('using resolved state value', () => {
+    it('should follow always transitions', () => {
+      const machine = createMachine({
+        initial: 'a',
+        states: {
+          a: {
+            always: [{ target: 'b' }]
+          },
+          b: {}
+        }
+      });
+
+      const service = interpret(machine).start(machine.getInitialState('a'));
```

Comment:
Interestingly, this doesn't follow always transitions:
```diff
-      const service = interpret(machine).start(machine.getInitialState('a'));
+      const service = interpret(machine).start('a');
```

I would consider this to be a bug too but it just never worked correctly and I don't think it's necessary to fix this in v4.